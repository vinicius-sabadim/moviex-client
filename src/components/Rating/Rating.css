.rating__container {
  bottom: 10px;
  display: flex;
  position: absolute;
  right: 10px;
}

.rating__block {
  align-items: center;
  display: flex;
  padding: 0 10px;
  position: relative;
}

.rating__rateContainer {
  margin-left: 8px;
  text-align: center;
}

.rating__rating {
  font-size: 20px;
  font-weight: bold;
}

.rating__total {
  display: block;
  font-size: 12px;
}
