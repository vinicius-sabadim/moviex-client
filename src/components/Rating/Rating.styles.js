import { css } from 'glamor'

export default {
  container: css({
    bottom: '20px',
    display: 'flex',
    position: 'absolute',
    right: '10px'
  }),
  block: css({
    alignItems: 'center',
    display: 'flex',
    padding: '0 10px'
  }),
  star: css({
    fontSize: '24px'
  }),
  rateContainer: css({
    marginLeft: '8px',
    textAlign: 'center'
  }),
  rating: css({
    fontSize: '20px',
    fontWeight: 'bold'
  }),
  total: css({
    display: 'block',
    fontSize: '12px'
  })
}
