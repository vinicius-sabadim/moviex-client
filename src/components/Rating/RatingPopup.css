.ratingPopup__container {
  background-color: #fff;
  border: 1px solid #ccc;
  box-shadow: 0 0 10px #ccc;
  padding: 10px;
  position: absolute;
  right: 5px;
  text-align: center;
  top: -70px;
  width: 360px;
}

.ratingPopup__buttonContainer {
  border-top: 1px solid #ccc;
  margin-top: 15px;
  padding-top: 10px;
}

.ratingPopup__button {
  background-color: #fff;
  border: none;
  border-bottom: 2px solid transparent;
  cursor: pointer;
  font-size: 16px;
  margin: 0 3px;
  padding: 5px;
}

.ratingPopup__button-save {
  color: var(--secondary-color);
}

.ratingPopup__button-save:hover {
  border-bottom: 2px solid var(--secondary-color);
}

.ratingPopup__button-cancel {
  color: #aaa;
  font-size: 14px;
}

.ratingPopup__button-cancel:hover {
  border-bottom: 2px solid #aaa;
}
