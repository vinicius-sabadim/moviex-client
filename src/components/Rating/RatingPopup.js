import React from 'react'

import Star from './Star'

import styles from './RatingPopup.styles'

const RatingPopup = () => {
  return (
    <div className={styles.container}>
      {[...Array(10)].map((_, i) => (
        <Star key={i} />
      ))}
    </div>
  )
}

export default RatingPopup
