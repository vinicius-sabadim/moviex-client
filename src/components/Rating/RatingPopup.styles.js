import { css } from 'glamor'

export default {
  container: css({
    backgroundColor: '#FFF',
    border: '1px solid #CCC',
    padding: '10px',
    position: 'absolute',
    right: '5px',
    textAlign: 'center',
    top: '-50px',
    width: '360px'
  })
}
