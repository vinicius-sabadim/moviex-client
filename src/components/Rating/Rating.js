import React, { useState } from 'react'

import RatingPopup from './RatingPopup'
import Star from '../Star'

import styles from './Rating.styles'

const Rating = () => {
  const [showPopup, setShowPopup] = useState(false)

  return (
    <div className={styles.container}>
      <div className={styles.block}>
        <Star color='#FCCA46' />
        <div className={styles.rateContainer}>
          <span className={styles.rating}>9.8</span>
          <span className={styles.total}>146</span>
        </div>
      </div>
      <div
        id="rate-this-toggle"
        className={styles.block}
        style={{ borderLeft: '1px solid #ccc', cursor: 'pointer' }}
        onClick={() => setShowPopup(true)}
      >
        <Star color='#ccc' />
        <div className={styles.rateContainer}>
          <span className={styles.total}>Rate this</span>
        </div>
      </div>
      {showPopup && <RatingPopup onCancel={() => setShowPopup(false)} />}
    </div>
  )
}

export default Rating
