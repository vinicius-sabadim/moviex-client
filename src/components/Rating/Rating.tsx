import React, { useState } from 'react'

import RatingPopup from './RatingPopup'
import Star from '../Star'

import Rate from '../../types/Rate'

import './Rating.css'

const Rating = () => {
  const [showPopup, setShowPopup] = useState(false)

  const handleSave = (rate: Rate) => {
    console.log(rate)
    setShowPopup(false)
  }

  return (
    <div className="rating__container">
      <div className="rating__block">
        <Star color="#FCCA46" />
        <div className="rating__rateContainer">
          <span className="rating__rating">9.8</span>
          <span className="rating__total">146</span>
        </div>
      </div>
      <div
        id="rate-this-toggle"
        className="rating__block"
        style={{ borderLeft: '1px solid #ccc', cursor: 'pointer' }}
        onClick={() => setShowPopup(true)}
      >
        <Star color="#ccc" />
        <div className="rating__rateContainer">
          <span className="rating__total">Rate this</span>
        </div>
      </div>
      {showPopup && (
        <RatingPopup onCancel={() => setShowPopup(false)} onSave={handleSave} />
      )}
    </div>
  )
}

export default Rating
