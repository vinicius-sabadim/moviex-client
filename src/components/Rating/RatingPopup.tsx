import React, { useState } from 'react'

import Star from '../Star'

import './RatingPopup.css'

const RatingPopup = ({ onCancel, onSave }) => {
  const [rate, setRate] = useState(6)

  const onSaveClicked = () => {
    onSave(rate)
  }

  return (
    <div className="ratingPopup__container">
      {[...Array(10)].map((_, index) => (
        <Star
          key={index}
          color={index + 1 <= rate ? '#FCCA46' : '#ccc'}
          hasEffect={true}
          style={{
            margin: '0 3px'
          }}
          onClick={() => setRate(index + 1)}
        />
      ))}
      <div className="ratingPopup__buttonContainer">
        <button
          className="ratingPopup__button ratingPopup__button-save"
          onClick={onSaveClicked}
        >
          Save
        </button>
        <button
          className="ratingPopup__button ratingPopup__button-cancel"
          onClick={onCancel}
        >
          Cancel
        </button>
      </div>
    </div>
  )
}

export default RatingPopup
