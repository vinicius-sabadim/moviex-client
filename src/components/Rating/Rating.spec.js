import React from 'react'
import { shallow } from 'enzyme'
import Rating from './Rating'

describe('Rating', () => {
  test('should toggle the popup to rate when clicked on rate this', () => {
    const component = shallow(<Rating />)
    const toggleArea = component.find('#rate-this-toggle')

    toggleArea.simulate('click')
    expect(component).toMatchSnapshot()
    //expect(component.find('RatePopup').exists()).toBeTruthy()

    toggleArea.simulate('click')
    expect(component).toMatchSnapshot()
    //expect(component.find('RatePopup').exists()).toBeFalsy()
  })
})
