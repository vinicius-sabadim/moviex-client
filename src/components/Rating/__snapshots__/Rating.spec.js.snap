// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating should toggle the popup to rate when clicked on rate this 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Rating />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className={
            Object {
              "data-css-120budj": "",
            }
          }
        >
          <Star
            color="#FCCA46"
          />
          <div
            className={
              Object {
                "data-css-o9v930": "",
              }
            }
          >
            <span
              className={
                Object {
                  "data-css-1niqrjj": "",
                }
              }
            >
              9.8
            </span>
            <span
              className={
                Object {
                  "data-css-1j754c6": "",
                }
              }
            >
              146
            </span>
          </div>
        </div>,
        <div
          className={
            Object {
              "data-css-120budj": "",
            }
          }
          id="rate-this-toggle"
          onClick={[Function]}
          style={
            Object {
              "borderLeft": "1px solid #ccc",
              "cursor": "pointer",
            }
          }
        >
          <Star
            color="#ccc"
          />
          <div
            className={
              Object {
                "data-css-o9v930": "",
              }
            }
          >
            <span
              className={
                Object {
                  "data-css-1j754c6": "",
                }
              }
            >
              Rate this
            </span>
          </div>
        </div>,
        <RatingPopup />,
      ],
      "className": Object {
        "data-css-1g76i17": "",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Star
              color="#FCCA46"
            />,
            <div
              className={
                Object {
                  "data-css-o9v930": "",
                }
              }
            >
              <span
                className={
                  Object {
                    "data-css-1niqrjj": "",
                  }
                }
              >
                9.8
              </span>
              <span
                className={
                  Object {
                    "data-css-1j754c6": "",
                  }
                }
              >
                146
              </span>
            </div>,
          ],
          "className": Object {
            "data-css-120budj": "",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "color": "#FCCA46",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className={
                    Object {
                      "data-css-1niqrjj": "",
                    }
                  }
                >
                  9.8
                </span>,
                <span
                  className={
                    Object {
                      "data-css-1j754c6": "",
                    }
                  }
                >
                  146
                </span>,
              ],
              "className": Object {
                "data-css-o9v930": "",
              },
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "9.8",
                  "className": Object {
                    "data-css-1niqrjj": "",
                  },
                },
                "ref": null,
                "rendered": "9.8",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "146",
                  "className": Object {
                    "data-css-1j754c6": "",
                  },
                },
                "ref": null,
                "rendered": "146",
                "type": "span",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Star
              color="#ccc"
            />,
            <div
              className={
                Object {
                  "data-css-o9v930": "",
                }
              }
            >
              <span
                className={
                  Object {
                    "data-css-1j754c6": "",
                  }
                }
              >
                Rate this
              </span>
            </div>,
          ],
          "className": Object {
            "data-css-120budj": "",
          },
          "id": "rate-this-toggle",
          "onClick": [Function],
          "style": Object {
            "borderLeft": "1px solid #ccc",
            "cursor": "pointer",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "color": "#ccc",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <span
                className={
                  Object {
                    "data-css-1j754c6": "",
                  }
                }
              >
                Rate this
              </span>,
              "className": Object {
                "data-css-o9v930": "",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Rate this",
                "className": Object {
                  "data-css-1j754c6": "",
                },
              },
              "ref": null,
              "rendered": "Rate this",
              "type": "span",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className={
              Object {
                "data-css-120budj": "",
              }
            }
          >
            <Star
              color="#FCCA46"
            />
            <div
              className={
                Object {
                  "data-css-o9v930": "",
                }
              }
            >
              <span
                className={
                  Object {
                    "data-css-1niqrjj": "",
                  }
                }
              >
                9.8
              </span>
              <span
                className={
                  Object {
                    "data-css-1j754c6": "",
                  }
                }
              >
                146
              </span>
            </div>
          </div>,
          <div
            className={
              Object {
                "data-css-120budj": "",
              }
            }
            id="rate-this-toggle"
            onClick={[Function]}
            style={
              Object {
                "borderLeft": "1px solid #ccc",
                "cursor": "pointer",
              }
            }
          >
            <Star
              color="#ccc"
            />
            <div
              className={
                Object {
                  "data-css-o9v930": "",
                }
              }
            >
              <span
                className={
                  Object {
                    "data-css-1j754c6": "",
                  }
                }
              >
                Rate this
              </span>
            </div>
          </div>,
          <RatingPopup />,
        ],
        "className": Object {
          "data-css-1g76i17": "",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Star
                color="#FCCA46"
              />,
              <div
                className={
                  Object {
                    "data-css-o9v930": "",
                  }
                }
              >
                <span
                  className={
                    Object {
                      "data-css-1niqrjj": "",
                    }
                  }
                >
                  9.8
                </span>
                <span
                  className={
                    Object {
                      "data-css-1j754c6": "",
                    }
                  }
                >
                  146
                </span>
              </div>,
            ],
            "className": Object {
              "data-css-120budj": "",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "color": "#FCCA46",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className={
                      Object {
                        "data-css-1niqrjj": "",
                      }
                    }
                  >
                    9.8
                  </span>,
                  <span
                    className={
                      Object {
                        "data-css-1j754c6": "",
                      }
                    }
                  >
                    146
                  </span>,
                ],
                "className": Object {
                  "data-css-o9v930": "",
                },
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "9.8",
                    "className": Object {
                      "data-css-1niqrjj": "",
                    },
                  },
                  "ref": null,
                  "rendered": "9.8",
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "146",
                    "className": Object {
                      "data-css-1j754c6": "",
                    },
                  },
                  "ref": null,
                  "rendered": "146",
                  "type": "span",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Star
                color="#ccc"
              />,
              <div
                className={
                  Object {
                    "data-css-o9v930": "",
                  }
                }
              >
                <span
                  className={
                    Object {
                      "data-css-1j754c6": "",
                    }
                  }
                >
                  Rate this
                </span>
              </div>,
            ],
            "className": Object {
              "data-css-120budj": "",
            },
            "id": "rate-this-toggle",
            "onClick": [Function],
            "style": Object {
              "borderLeft": "1px solid #ccc",
              "cursor": "pointer",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "color": "#ccc",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <span
                  className={
                    Object {
                      "data-css-1j754c6": "",
                    }
                  }
                >
                  Rate this
                </span>,
                "className": Object {
                  "data-css-o9v930": "",
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Rate this",
                  "className": Object {
                    "data-css-1j754c6": "",
                  },
                },
                "ref": null,
                "rendered": "Rate this",
                "type": "span",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Rating should toggle the popup to rate when clicked on rate this 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Rating />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className={
            Object {
              "data-css-120budj": "",
            }
          }
        >
          <Star
            color="#FCCA46"
          />
          <div
            className={
              Object {
                "data-css-o9v930": "",
              }
            }
          >
            <span
              className={
                Object {
                  "data-css-1niqrjj": "",
                }
              }
            >
              9.8
            </span>
            <span
              className={
                Object {
                  "data-css-1j754c6": "",
                }
              }
            >
              146
            </span>
          </div>
        </div>,
        <div
          className={
            Object {
              "data-css-120budj": "",
            }
          }
          id="rate-this-toggle"
          onClick={[Function]}
          style={
            Object {
              "borderLeft": "1px solid #ccc",
              "cursor": "pointer",
            }
          }
        >
          <Star
            color="#ccc"
          />
          <div
            className={
              Object {
                "data-css-o9v930": "",
              }
            }
          >
            <span
              className={
                Object {
                  "data-css-1j754c6": "",
                }
              }
            >
              Rate this
            </span>
          </div>
        </div>,
        <RatingPopup />,
      ],
      "className": Object {
        "data-css-1g76i17": "",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Star
              color="#FCCA46"
            />,
            <div
              className={
                Object {
                  "data-css-o9v930": "",
                }
              }
            >
              <span
                className={
                  Object {
                    "data-css-1niqrjj": "",
                  }
                }
              >
                9.8
              </span>
              <span
                className={
                  Object {
                    "data-css-1j754c6": "",
                  }
                }
              >
                146
              </span>
            </div>,
          ],
          "className": Object {
            "data-css-120budj": "",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "color": "#FCCA46",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className={
                    Object {
                      "data-css-1niqrjj": "",
                    }
                  }
                >
                  9.8
                </span>,
                <span
                  className={
                    Object {
                      "data-css-1j754c6": "",
                    }
                  }
                >
                  146
                </span>,
              ],
              "className": Object {
                "data-css-o9v930": "",
              },
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "9.8",
                  "className": Object {
                    "data-css-1niqrjj": "",
                  },
                },
                "ref": null,
                "rendered": "9.8",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "146",
                  "className": Object {
                    "data-css-1j754c6": "",
                  },
                },
                "ref": null,
                "rendered": "146",
                "type": "span",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Star
              color="#ccc"
            />,
            <div
              className={
                Object {
                  "data-css-o9v930": "",
                }
              }
            >
              <span
                className={
                  Object {
                    "data-css-1j754c6": "",
                  }
                }
              >
                Rate this
              </span>
            </div>,
          ],
          "className": Object {
            "data-css-120budj": "",
          },
          "id": "rate-this-toggle",
          "onClick": [Function],
          "style": Object {
            "borderLeft": "1px solid #ccc",
            "cursor": "pointer",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "color": "#ccc",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <span
                className={
                  Object {
                    "data-css-1j754c6": "",
                  }
                }
              >
                Rate this
              </span>,
              "className": Object {
                "data-css-o9v930": "",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Rate this",
                "className": Object {
                  "data-css-1j754c6": "",
                },
              },
              "ref": null,
              "rendered": "Rate this",
              "type": "span",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className={
              Object {
                "data-css-120budj": "",
              }
            }
          >
            <Star
              color="#FCCA46"
            />
            <div
              className={
                Object {
                  "data-css-o9v930": "",
                }
              }
            >
              <span
                className={
                  Object {
                    "data-css-1niqrjj": "",
                  }
                }
              >
                9.8
              </span>
              <span
                className={
                  Object {
                    "data-css-1j754c6": "",
                  }
                }
              >
                146
              </span>
            </div>
          </div>,
          <div
            className={
              Object {
                "data-css-120budj": "",
              }
            }
            id="rate-this-toggle"
            onClick={[Function]}
            style={
              Object {
                "borderLeft": "1px solid #ccc",
                "cursor": "pointer",
              }
            }
          >
            <Star
              color="#ccc"
            />
            <div
              className={
                Object {
                  "data-css-o9v930": "",
                }
              }
            >
              <span
                className={
                  Object {
                    "data-css-1j754c6": "",
                  }
                }
              >
                Rate this
              </span>
            </div>
          </div>,
          <RatingPopup />,
        ],
        "className": Object {
          "data-css-1g76i17": "",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Star
                color="#FCCA46"
              />,
              <div
                className={
                  Object {
                    "data-css-o9v930": "",
                  }
                }
              >
                <span
                  className={
                    Object {
                      "data-css-1niqrjj": "",
                    }
                  }
                >
                  9.8
                </span>
                <span
                  className={
                    Object {
                      "data-css-1j754c6": "",
                    }
                  }
                >
                  146
                </span>
              </div>,
            ],
            "className": Object {
              "data-css-120budj": "",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "color": "#FCCA46",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className={
                      Object {
                        "data-css-1niqrjj": "",
                      }
                    }
                  >
                    9.8
                  </span>,
                  <span
                    className={
                      Object {
                        "data-css-1j754c6": "",
                      }
                    }
                  >
                    146
                  </span>,
                ],
                "className": Object {
                  "data-css-o9v930": "",
                },
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "9.8",
                    "className": Object {
                      "data-css-1niqrjj": "",
                    },
                  },
                  "ref": null,
                  "rendered": "9.8",
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "146",
                    "className": Object {
                      "data-css-1j754c6": "",
                    },
                  },
                  "ref": null,
                  "rendered": "146",
                  "type": "span",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Star
                color="#ccc"
              />,
              <div
                className={
                  Object {
                    "data-css-o9v930": "",
                  }
                }
              >
                <span
                  className={
                    Object {
                      "data-css-1j754c6": "",
                    }
                  }
                >
                  Rate this
                </span>
              </div>,
            ],
            "className": Object {
              "data-css-120budj": "",
            },
            "id": "rate-this-toggle",
            "onClick": [Function],
            "style": Object {
              "borderLeft": "1px solid #ccc",
              "cursor": "pointer",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "color": "#ccc",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <span
                  className={
                    Object {
                      "data-css-1j754c6": "",
                    }
                  }
                >
                  Rate this
                </span>,
                "className": Object {
                  "data-css-o9v930": "",
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Rate this",
                  "className": Object {
                    "data-css-1j754c6": "",
                  },
                },
                "ref": null,
                "rendered": "Rate this",
                "type": "span",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
